prisma-fmt/src/code_actions.rs

```diff
@@ -3,89 +3,128 @@ mod multi_schema;
 mod relation_mode;
 mod relations;
 
-use lsp_types::{CodeActionOrCommand, CodeActionParams, Diagnostic, Range, TextEdit, WorkspaceEdit};
+use log::warn;
+use lsp_types::{CodeActionOrCommand, CodeActionParams, Diagnostic, Range, TextEdit, Url, WorkspaceEdit};
 use psl::{
-    diagnostics::Span,
+    diagnostics::{FileId, Span},
     parser_database::{
         ast,
         walkers::{ModelWalker, RefinedRelationWalker, ScalarFieldWalker},
-        SourceFile,
+        ParserDatabase, SourceFile,
     },
     schema_ast::ast::{Attribute, IndentationType, NewlineType, WithSpan},
-    PreviewFeature,
+    Configuration, Datasource, PreviewFeature,
 };
-use std::{collections::HashMap, sync::Arc};
+use std::collections::HashMap;
+
+pub(super) struct CodeActionsContext<'a> {
```

Comment:
Bunch of stuff that almost every code action ended up needing, it is easier to pass around when it is grouped together.